baybe/objective.py

```diff
@@ -51,6 +51,21 @@ class Objective(SerialMixin):
     )
     """The function used to combine the different targets."""
 
+    def __str__(self) -> str:
+        start_bold = "\033[1m"
+        end_bold = "\033[0m"
+
+        # Convert the targets list to a dataframe to have a tabullar output
+        targets_list = [target.summary() for target in self.targets]
+        targets_df = pd.DataFrame(targets_list)
+        targets_df["Weight"] = self.weights
+
+        objective_str = f"""{start_bold}Objective{end_bold}
+        \n{start_bold}Mode: {end_bold}{self.mode}
+        \n{start_bold}Targets {end_bold}\n{targets_df}
+        \n{start_bold}Combine Function: {end_bold}{self.combine_func}"""
+        return objective_str.replace("\n", "\n ").replace("\r", "\r ")
```

Comment:
why are you adding a space here at this level? if I print an object (and not an entire searchspace or campaign) I'd expect no space-indent , so shouldnt the space be added on levels above that receive this string?

Reply:
It will add only one space in case you want to print the objective object, like this: 
![image](https://github.com/emdgroup/baybe/assets/118910837/03407b20-dae3-4b89-bac7-7c3d373529bd)
And this is really needed for the campaign hirachical output, because in the campaign I replaced `\n` by `\n ` which will aply to the Searchspce, objective and recommenders but not to the sub-spaces. In case I remove the replace part it will return the following output (even in the campaign print): 
![image](https://github.com/emdgroup/baybe/assets/118910837/7949028d-0138-4b7a-95ad-8b386a79cdf6)
But I will remove the `\r` bcause it is not used here.

Reply:
makes sense